fe/fe-core/src/main/java/org/apache/doris/statistics/Histogram.java

```diff
@@ -0,0 +1,180 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.statistics;
+
+import org.apache.doris.catalog.PrimitiveType;
+import org.apache.doris.catalog.Type;
+import org.apache.doris.statistics.util.StatisticsUtil;
+
+import com.alibaba.fastjson2.JSON;
+import com.alibaba.fastjson2.JSONArray;
+import com.alibaba.fastjson2.JSONObject;
+import com.google.common.collect.Lists;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.parquet.Strings;
+
+import java.util.List;
+
+public class Histogram {
+    private static final Logger LOG = LogManager.getLogger(Histogram.class);
+
+    private Type dataType;
+
+    private int maxBucketSize;
```

Comment:
Is bucketSize means how many values in each buckets? if so, according to the documents, the patch implements a equal-hight historam, then what's the meaning of max bucket size? In my humble opinion, the value of bucket size of such histogram is same among all the buckets

Reply:
This actually refers to the maximum number of buckets allowed to be divided into (that is, the upper limit of the number of buckets), not the width of the bucket.